File: packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/MediaDecryption.m
// MediaDecryption.m
#import "MediaDecryption.h"

@implementation MediaDecryption

- (instancetype)initWithUid:(NSString *)uid
                            iat:(NSInteger)iat
                             dk:(NSData *)dk
                             iv:(NSData *)iv
                        dkAesIv:(NSData *)dkAesIv
                        ivAesIv:(NSData *)ivAesIv {
    self = [super init];
    if (self) {
        _uid = uid;
        _iat = iat;
        _dk = dk;
        _iv = iv;
        _dkAesIv = dkAesIv;
        _ivAesIv = ivAesIv;
    }
    return self;
}

@end

File: packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/MediaDecryption.java
package io.flutter.plugins.videoplayer;

import androidx.annotation.Nullable;

public class MediaDecryption {
    public String uid;
    public int iat;
    public final byte[] dk;
    public final byte[] dkAesIv;
    public final byte[] iv;
    public final byte[] ivAesIv;

    public MediaDecryption(String uid, int iat, @Nullable byte[] dk, @Nullable byte[] dkAesIv,
                           @Nullable byte[] iv, @Nullable byte[] ivAesIv) {
        this.uid = uid;
        this.iat = iat;
        this.dk = dk;
        this.dkAesIv = dkAesIv;
        this.iv = iv;
        this.ivAesIv = ivAesIv;
    }
}
